src/pages/predictions.vue

```diff
@@ -0,0 +1,416 @@
+                localStorage.setItem(
+                  "starred",
+                  JSON.stringify(starredCountries)
+                );
+                $("#chart-star-" + index + "").removeClass("btn-star-active");
+                $("#fa-star-" + index + "").removeClass("star-color");
+              } else {
+                starredCountries.push(i.country);
+                localStorage.setItem(
+                  "starred",
+                  JSON.stringify(starredCountries)
+                );
+              }
+            }
+          } else {
+            console.log(`Your browser doesn't support Web Storage!`);
+          }
+        });
+    },
+    userFunctions: function() {
+      this.sortedPredictedArray.forEach((i, index) => {
+        this.starredButtonOnClick(i, index);
+      });
+      this.searchCountries();
+      this.ifAllCountriesBtnClicked();
+      this.ifStarredBtnClicked();
+    },
+    searchCountries: function() {
+      $("#results-not-found").hide();
+      $("#search").keyup(function() {
+        var text = $(this)
+          .val()
+          .toLowerCase();
+        if (text.length === 0) {
+          $("#tab-choose .btn-country").click();
+        }
+        $(".content").hide();
+        var resultCount = 0;
+        $("#results-not-found").hide();
+        $(".content .country-names").each(function() {
+          if (
+            $(this)
+              .text()
+              .toLowerCase()
+              .indexOf("" + text + "") != -1
+          ) {
+            $(this)
+              .closest(".content")
+              .show();
+            $("#results-not-found").hide();
+            resultCount++;
+          }
+
+          if (resultCount == 0) {
+            $("#results-not-found").show();
+          }
+        });
+      });
+    },
+    ifAllCountriesBtnClicked: function() {
+      let that = this;
+      $("#tab-choose .btn-country").click(function() {
+        $("#tab-choose .btn-starred").removeClass("active");
+        $("#tab-choose .btn-country").addClass("active");
+        $(".content").show();
+        $("#starred-none").hide();
+        $("#results-not-found").hide();
+        that.sortedPredictedArray.forEach((i, index) => {
+          if (typeof Storage !== "undefined") {
+            if (localStorage.getItem("starred") === null) {
+              //
+            } else {
+              let starredCountriesGenAddress = JSON.parse(
+                localStorage.getItem("starred")
+              );
+
+              if (starredCountriesGenAddress.includes(i.country)) {
+                $("#chart-star-" + index + "")
+                  .removeClass("inactive")
+                  .addClass("btn-star-active");
+                $("#fa-star-" + index + "").addClass("star-color");
+              }
+            }
+          } else {
+            console.log(`Your browser doesn't support Web Storage!`);
+          }
+        });
+      });
+    },
+    ifStarredBtnClicked: function() {
```

Comment:
maintain a separate array in local storage to star/un-star prediction graphs. This should work as a standalone feature and not interfere with the main graphs. I would also suggest you write a helper function in a constants file and pass arguments to identify which starred(normal/predictions) is being called. This way we don't have to write the same function twice.